### PR TITLE
Fix centipede crash parser

### PR DESCRIPTION
**Pull request by Alan32Liu**
To adapt to recent changes in `Centipede`'s log output, this PR:
1. Edits reproducer parser RegEx.
2. Removes the redundant `'CRASH LOG:'` prefix
3. Updates `Centipede` binary for testcases
4. Modifies unittests to capture 1 & 2.

### CODE REVIEW

**Comment by Alan32Liu** [comment]
Pass unittests locally with `INTEGRATION=1` enabled.

File: src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py [hunk 1]
```diff
@@ -39,7 +39,8 @@
     f'--address_space_limit_mb={_ADDRESS_SPACE_LIMIT}',
 ]
 
-_CRASH_REGEX = re.compile(r'Crash detected, saving input to (.*)')
+_CRASH_REGEX = re.compile(r'Saving input to: [\n]?(.*)')
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py:42]: Can we support multiple formats here, via `|` in our regex pattern?

File: src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py [hunk 2]
```diff
@@ -39,7 +39,8 @@
...
+_REDUNDANT_PREFIX = 'CRASH LOG: '
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py:43]: nit: rename this to CRASH_LOG_PREFIX.

**Comment by Alan32Liu** [comment]
Cloud build failure due to [using `/lib/x86_64-linux-gnu/libstdc++.so.6`](https://pantheon.corp.google.com/cloud-build/builds/42157685-5a71-4a16-9420-c8d3ba3f9776;step=4?project=clusterfuzz-testing-public), investigating this now.

File: src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py [hunk 3]
```diff
@@ -152,6 +154,18 @@ def fuzz(self, target_path, options, reproducers_dir, max_time):  # pylint: disa
     return engine.FuzzResult(fuzz_result.output, fuzz_result.command, crashes,
                              stats, fuzz_result.time_executed)
 
+  def _trim_logs(self, fuzz_result):
+    """ Trims the fuzzer log for stacktrace parser.
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py:158]: Nit: add more comments here on why this is needed.

```
The CRASH LOG: prefix breaks our stacktrace parsing, so we strip it.
```

File: src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py [hunk 4]
```diff
@@ -138,6 +139,7 @@ def fuzz(self, target_path, options, reproducers_dir, max_time):  # pylint: disa
     timeout = max_time + _CLEAN_EXIT_SECS
     fuzz_result = runner.run_and_wait(
         additional_args=arguments, timeout=timeout)
+    self._trim_logs(fuzz_result)
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py:142]: I think we need this for every other method as well that produces output? e.g. `reproduce`?

**Review by oliverchang — APPROVED** [review]

**Pull request merged**